_episodes/31.md

```diff
@@ -0,0 +1,579 @@
+
+Trino 366
+
+* Support for automatic query retries
+* Support for `DENY` security rules
+* Performance optimizations
+
+Release notes: <https://trino.io/docs/current/release/release-366.html>
+
+Download: <https://trino.io/download.html>
+
+Manfred's additional notes:
+
+* Cool new SQL like `TRUNCATE TABLE` and support for time travel 
+* `contains` function for IP check in CIDR
+* Lots of performance and correctness fixes on Hive and Iceberg connectors
+* Drop support for old Pinot versions
+* Support for Hive to Iceberg redirects
+* Automatic TLS for internal communication support for Java 17
+
+And a last note, full Java 17 support is becoming a reality.
+
+More detailed information is available in the [365](https://trino.io/docs/current/release/release-365.html)
+and [366](https://trino.io/docs/current/release/release-366.html) release notes.
+
+## Log4Shell
+
+There's a new vulnerability in town that has the potential to affect Java
+projects that use some Log4j2 versions. It is called Log4Shell, and it does not
+affect Trino. Read [the blog for more details](https://trino.io/blog/2021/12/13/log4shell-does-not-affect-trino.html).
+
+<p align="center">
+ <img align="center" width="50%" src="/assets/episode/31/log4shell.jpeg"/><br/>
+</p>
+
+## Concept of the month: ReplicaSets, Deployments, and Services
+
+In [the first installment of Trinetes](/episodes/24.html), we covered what 
+containerization is and why we use it. We covered the difference between tools
+like docker-compose and container orchestration systems like Kubernetes (k8s).
+Finally, we covered the first k8s object called a [_pod_](https://kubernetes.io/docs/concepts/workloads/pods/).
+
+As a reminder, a pod is the basic unit of deployment in a k8s cluster. In this
+episode, we cover how to scale, deploy, and connect these Pods. If you are 
+missing some context, you should review [the first installment of this series](/episodes/24.html). 
+After we cover some new k8s objects, we cover the steps 
+
+### Replica Sets
+
+_Replicas_ make one or more instances based on the same pod definitions. In k8s,
+the object used to manage replication is a [_ReplicaSet_](https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/).
+
+ReplicaSets provide high availability by making multiple instances based on a 
+pod definition in the k8s cluster. Kubernetes automatically brings up any failed
+pod instances that go down in a ReplicaSets based on the number of replicas you
+specify in the definition.
+
+Replication also enables load balancing IO traffic over multiple pods. You then
+gain the flexibility to scale up or down as traffic increases or decreases
+without any downtime.
+
+To scale the number of pods in a live ReplicaSet, you can update the `replicas` 
+value in the ReplicaSet definition file then running the following command to
+update it:
+
+```
+kubectl replace -f replicaset-definition.yml
+```
+
+You can also edit the live ReplicaSet without changing the local file:
```

Comment:
Probably should decide if you always use the code name "ReplicaSet" or imho better just treat is as normal words "replica sets". And only use the camel case when it is specifically about code..

Reply:
I'll go with always using it for simplicity.